include/gzip/decompress.hpp

```diff
@@ -90,8 +91,7 @@ std::string decompress(const char * data, std::size_t size)
 {
     Decompressor decomp;
     std::string output;
-    std::size_t uncompressed_size = decomp.decompress(output,data,size);
-    output.resize(uncompressed_size);
+    decomp.decompress(output,data,size);
```

Comment:
The resize is done here rather than in the lower level API to avoid the cost of the resize in situations where you can pass the buffer along to an API that accepts a separate size argument. So, for a downstream usage of the buffer than only accepts a string, you'd obviously need to resize, but for an api that accepts `const char *, std::size_t` you don't need to resize and can use the buffer as it, and pass `size_uncompressed` to avoid the resize (and its cost).

Reply:
@springmeyer resize is required otherwise the length of the string will be too long. It will be an invalid decompression. By default resize() will fill new characters in the loop prior to the default value for `new char()`. This actually extends the size of the buffer in the start of our decompression loop, which is required as the data must be preallocated and assigned in a std::string so it is not writing to potential areas larger then `size()`. However, after we are done we may not have used all the space that was allocated and assigned. This will leave an extended set of data on our buffer (set to `new char()`) that is no a valid part of our decompressed data. This must be trimmed down with our final `resize()`.

Reply:
👌